src/main/java/co/cask/hydrator/plugin/RedshiftToS3Action.java

```diff
@@ -0,0 +1,308 @@
+    StringBuilder unloadCommand = new StringBuilder();
+    unloadCommand.append("unload ('").append(config.query).append("') to '").append(config.s3DataPath).append("'");
+    // Check authentication is using keys or role.
+    if (!Strings.isNullOrEmpty(config.accessKey) && !Strings.isNullOrEmpty(config.secretAccessKey)) {
+      unloadCommand.append(" ");
+      unloadCommand.append("credentials 'aws_access_key_id=").append(config.accessKey).append(";aws_secret_access_key=")
+        .append(config.secretAccessKey).append("'");
+    } else {
+      unloadCommand.append(" iam_role '").append(config.iamRole).append("'");
+    }
+    if (!Strings.isNullOrEmpty(config.delimiter)) {
+      unloadCommand.append(" delimiter '").append(config.delimiter).append("'");
+    }
+    if (!config.parallel) {
+      unloadCommand.append(" parallel off");
+    }
+    if (config.manifest) {
+      unloadCommand.append(" manifest");
+    }
+    if (config.allowOverWrite) {
+      unloadCommand.append(" allowoverwrite");
+    }
+    if (config.addQuotes) {
+      unloadCommand.append(" addquotes");
+    }
+    if (config.escape) {
+      unloadCommand.append(" escape");
+    }
+    String compression = config.compression;
+    if (!Strings.isNullOrEmpty(compression)) {
+      switch (config.compression.toUpperCase()) {
+        case COMPRESSION_BZIP2:
+          unloadCommand.append(" ").append(COMPRESSION_BZIP2);
+          break;
+        case COMPRESSION_GZIP:
+          unloadCommand.append(" ").append(COMPRESSION_GZIP);
+          break;
+        case COMPRESSION_NONE:
+          break;
+        default:
+          throw new IllegalArgumentException("Unsupported compression type " + config.compression);
+      }
+    }
+    unloadCommand.append(";");
+    return unloadCommand.toString();
+  }
+
+  /**
+   * Config class that contains all properties required for running the unload command.
+   */
+  public static class RedshiftToS3Config extends PluginConfig {
+    @Macro
+    @Nullable
+    @Description("The access Id provided by AWS to access the S3 bucket. Both configurations 'Keys(Access and Secret " +
+      "Access keys)' and 'IAM Role' can not be provided or empty at the same time. (Macro-enabled)")
+    private String accessKey;
+
+    @Macro
+    @Nullable
+    @Description("AWS secret access key having access to the S3 bucket. Both configurations 'Keys(Access and Secret " +
+      "Access keys)' and 'IAM Role' can not be provided or empty at the same time.(Macro-enabled)")
+    private String secretAccessKey;
+
+    @Macro
+    @Nullable
+    @Description("IAM role having GET,LIST and PUT permissions to the S3 bucket. Both configurations 'Keys(Access " +
+      "and Secret Access keys)' and 'IAM Role' can not be provided or empty at the same time.(Macro-enabled)")
+    private String iamRole;
+
+    @Description("A SELECT query, the results of which query are unloaded from Redshift table to the S3 bucket.")
+    private String query;
```

Comment:
query was not kept macro-enabled, since change in query can alter the schema of the data being unloaded.
However, changed query, s3DataPath and all the related commands for unload (delimiter, compression, etc) to macro in the latest commit.